ampd/src/broadcaster_v2/msg_queue.rs

```diff
@@ -117,23 +119,32 @@ where
     /// * `Error::EnqueueMsg` - If enqueueing fails
     /// * `Error::GasExceedsGasCap` - If the message requires more gas than allowed
     /// * `Error::ReceiveTxResult` - If the result channel is closed prematurely
-    pub async fn enqueue(
-        &mut self,
-        msg: Any,
-    ) -> Result<impl Future<Output = Result<(String, u64)>> + Send> {
-        let attachment = json!({ "msg": &msg });
+    #[instrument(skip(self))]
+    pub async fn enqueue(&mut self, msg: Any) -> Result<impl Future<Output = TxResult> + Send> {
         let rx = self
-            .enqueue_with_channel(msg)
+            .enqueue_with_channel(msg.clone())
             .await
-            .map_err(|err| err.attach_printable(attachment.clone()))?;
+            .inspect_err(|err| {
+                error!(
+                    err = LoggableError::from(err).as_value(),
+                    msg = ?msg,
```

Comment:
the msg is now captured through `#[instrument]`, so I think we can remove it here and stop cloning